dyninstAPI/src/BPatch_binaryEdit.C

```diff
@@ -87,7 +87,7 @@ BPatch_binaryEdit::BPatch_binaryEdit(const char *path, bool openDependencies) :
 
   startup_printf("[%s:%d] - Opening original file %s\n",
                  FILE__, __LINE__, path);
-  origBinEdit = BinaryEdit::openFile(std::string(path));
+  origBinEdit = BinaryEdit::openFile(path ? path : "");
```

Comment:
Should we do the 
if(path == nullptr) return 
or something like the next commit?

Reply:
>Should we do the
>if(path == nullptr) return

Since the next like checks the success of `openFile` _and_ `openFile` will fail for an invalid path, then this change at least avoids throwing an exception when `path` is null.

>or something like the next commit?

The early return from `BPatch::openBinary` is fine because it's a (sort-of) factory function, so errors can be expected. A constructor should never leave an object in a partially-constructed state. If I had the time, I'd rip out all of that dangerous nonsense in these BPatch classes.